### DISCUSSION TITLE
500 server error - No logs

### DISCUSSION THREAD

**Discussion by dlionel** [deta/cloud-docs#456] [Help] [answered]
<img width="990" alt="image" src="https://user-images.githubusercontent.com/7291417/188318018-55b050ce-29e4-4ef1-a93e-994697deca6a.png">

Hello, 

Great work so far on Deta! Just getting started. 

I am concerned by the fact that I am getting a 500 server error but there is "no errors" in the log when checking with visor. No luck either using the command `deta logs --follow` from a terminal. 

My question is why is the error not logged anywhere? Am I missing something?

Thanks,

**Comment by jnsougata** [comment]
There’s nothing you can do… it’s your code issue… try running the server locally to debug it

**Comment by dlionel** [comment]
Okay, thank you for your input. But it is working fine locally. I can pinpoint the issue by commenting some part of the code. Although, I believe a platform that want to be accessible should always be able to mange error logging when things go wrong.

**Comment by abdelhai** [comment]
use `deta logs --follow`

**Comment by jnsougata** [comment, marked as answer]
this gist might help you too: https://gist.github.com/jnsougata/e835e6c914278853af35331b28bdf163

**Comment by dlionel** [comment]
Great thank you. I am going to try it.

**Comment by dlionel** [comment]
I finally created a re-usable middleware using the code on the gist 

```
import traceback
import typing
from starlette.middleware.base import BaseHTTPMiddleware
from starlette.responses import JSONResponse

class LogServerErrorMiddleware(BaseHTTPMiddleware):
    """Return the server error response in a JSON object for debugging.

    Make sure that this middleware is not enable in production
    as it could leak important security information.
    """

    async def dispatch(
        self, request: Request, call_next: typing.Callable[[Request], typing.Awaitable[Response]]
    ) -> Response:
        """Render server error."""
        try:
            return await call_next(request)
        except Exception as exc:
            logger.exception(exc)
            trace = traceback.format_exception(type(exc), exc, exc.__traceback__)
            return JSONResponse(content={"traceback": trace}, status_code=500)
```

And then in my main file
```
if AppSettings.APP_ENV != "PROD":
    app.add_middleware(LogServerErrorMiddleware)
```
